compiler/record-minmax/include/MinMaxObserver.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __RECORD_MINMAX_MINMAXOBSERVER_H__
+#define __RECORD_MINMAX_MINMAXOBSERVER_H__
+
+#include <luci_interpreter/Interpreter.h>
+#include <luci_interpreter/core/Tensor.h>
+
+#include <vector>
+#include <unordered_map>
+
+namespace record_minmax
+{
+
+struct MinMaxVectors
+{
+  std::vector<float> min_vector;
+  std::vector<float> max_vector;
+};
+
+class MinMaxMap
+{
+public:
+  // Record min/max of node
+  void recordMinMax(const luci::CircleNode *node, float min, float max)
+  {
+    const auto iter = _minmax_map.find(node);
+    if (iter == _minmax_map.end())
+    {
+      MinMaxVectors vectors;
+      vectors.min_vector.push_back(min);
+      vectors.max_vector.push_back(max);
+      _minmax_map.emplace(node, vectors);
+    }
+    else
+    {
+      auto vectors = iter->second;
+      vectors.min_vector.push_back(min);
+      vectors.max_vector.push_back(max);
+    }
```

Comment:
`operator[]` will insert an element if it is missing:

```suggestion
    MinMaxVectors& vectors = _minmax_map[node];
    vectors.min_vector.push_back(min);
    vectors.max_vector.push_back(max);
```

or

```suggestion
    _minmax_map[node].min_vector.push_back(min);
    _minmax_map[node].max_vector.push_back(max);
```

Reply:
Thanks for the suggestions. I've accepted the first one.